app/uk/gov/hmrc/tai/connectors/BbsiConnector.scala

```diff
@@ -16,33 +16,54 @@
 
 package uk.gov.hmrc.tai.connectors
 
-import com.google.inject.{Inject, Singleton}
+import java.util.UUID.randomUUID
 
-import scala.concurrent.{ExecutionContext, Future}
+import com.google.inject.{Inject, Singleton}
+import com.kenshoo.play.metrics.Metrics
+import com.typesafe.scalalogging.LazyLogging
 import uk.gov.hmrc.domain.Nino
-import uk.gov.hmrc.tai.model.domain.BankAccount
-import uk.gov.hmrc.tai.model.tai.TaxYear
-import uk.gov.hmrc.tai.model.enums.APITypes
+import uk.gov.hmrc.http.HttpReads.Implicits.readRaw
 import uk.gov.hmrc.http._
+import uk.gov.hmrc.http.logging.SessionId
+import uk.gov.hmrc.play.bootstrap.http.HttpClient
 import uk.gov.hmrc.tai.config.DesConfig
-import uk.gov.hmrc.tai.model.domain.formatters.BbsiHodFormatters
+import uk.gov.hmrc.tai.metrics.metrics.HasMetrics
+import uk.gov.hmrc.tai.model.tai.TaxYear
 import uk.gov.hmrc.tai.util.TaiConstants
 
+import scala.concurrent.{ExecutionContext, Future}
+
 @Singleton
-class BbsiConnector @Inject()(httpHandler: HttpHandler, urls: BbsiUrls, config: DesConfig)(
-  implicit ec: ExecutionContext) {
-
-  def createHeader: HeaderCarrier =
-    HeaderCarrier(
-      extraHeaders = Seq(
-        "Environment"   -> config.environment,
-        "Authorization" -> s"Bearer ${config.authorization}",
-        "Content-Type"  -> TaiConstants.contentType))
-
-  def bankAccounts(nino: Nino, taxYear: TaxYear)(implicit hc: HeaderCarrier): Future[Seq[BankAccount]] = {
-    implicit val hc: HeaderCarrier = createHeader
-    httpHandler.getFromApi(urls.bbsiUrl(nino, taxYear), APITypes.BbsiAPI) map { json =>
-      json.as[Seq[BankAccount]](BbsiHodFormatters.bankAccountHodReads)
+class BbsiConnector @Inject()(val metrics: Metrics, http: HttpClient, urls: BbsiUrls, config: DesConfig)(
+  implicit ec: ExecutionContext)
+    extends LazyLogging with HasMetrics {
+
+  private def correlationId(hc: HeaderCarrier): String = {
+    val CorrelationIdPattern = """.*([A-Za-z0-9]{8}-[A-Za-z0-9]{4}-[A-Za-z0-9]{4}-[A-Za-z0-9]{4}).*""".r
+    hc.requestId match {
+      case Some(requestId) =>
+        requestId.value match {
+          case CorrelationIdPattern(prefix) => prefix + "-" + randomUUID.toString.substring(24)
+          case _                            => randomUUID.toString
+        }
+      case _ => randomUUID.toString
     }
```

Comment:
Sorry this is obsolete now. The request id is going to change and won't be UUID anymore. Just use randomUUID.toString without deriving it from request_id